### PR TITLE
Align tokenizers in pipeline and api_server benchmark scripts

### PR DESCRIPTION
**Pull request by AllentDan**
#1642

### CODE REVIEW

File: lmdeploy/tokenizer.py [hunk 1]
```diff
@@ -572,15 +576,15 @@ def detokenize_incrementally(self,
             skip_special_tokens=skip_special_tokens,
             spaces_between_special_tokens=spaces_between_special_tokens)
 
-    def __call__(self, s: Union[str, Sequence[str]]):
+    def __call__(self, s: Union[str, Sequence[str]], *args, **kwargs):
```
Comment (AllentDan) [inline lmdeploy/tokenizer.py:579]: A little weird, since it does not call `self.encode` directly.

File: lmdeploy/tokenizer.py [hunk 2]
```diff
@@ -181,10 +185,11 @@ def __call__(self, s: Union[str, Sequence[str]]):
...
         """
         import addict
-        add_bos = False
-        add_eos = False
 
-        input_ids = self.model.Encode(s, add_bos=add_bos, add_eos=add_eos)
+        input_ids = self.model.Encode(s,
+                                      add_bos=add_bos,
+                                      add_eos=add_eos,
+                                      **kwargs)
```
Comment (lvhan028) [inline lmdeploy/tokenizer.py:192]: What's the difference from previous version?

**Comment by lvhan028** [comment]
I think the straightforward way is to use AutoTokenizer in profile_restful_api.py, isn't it?

**Comment by AllentDan** [comment]
> I think the straightforward way is to use AutoTokenizer in profile_restful_api.py, isn't it?

Yes. But I don't know if it looks good that we keep `__call__` function using `add_special_tokens==False`

**Comment by lvhan028** [comment]
> > I think the straightforward way is to use AutoTokenizer in profile_restful_api.py, isn't it?
> 
> Yes. But I don't know if it looks good that we keep `__call__` function using `add_special_tokens==False`

Is Tokenizer's __call__ being invoked by other module besides the benchmark script?

**Comment by AllentDan** [comment]
> I think the straightforward way is to use AutoTokenizer in profile_restful_api.py, isn't it?

Actually, only `benchmark/profile_pipeline_api.py` and `benchmark/benchmark_decode.py` use AutoTokenizer. So, all scripts Tokenizer or AutoToeknizer?

**Comment by lvhan028** [comment]
I think we can use AutoTokenizer in the benchmark script

**Comment by lvhan028** [comment]
I still want to keep Tokenizer.__call__, as AutoTokenizer has the same attribute.

File: lmdeploy/tokenizer.py (hunk 2 above)
Comment (AllentDan) [reply lmdeploy/tokenizer.py:192]: previous version does not passing kwargs in.

**Comment by lvhan028** [comment]
I did get the title "Set the add_special_token default to True"
Is add_special_token==True indicated by *args and **kwargs?

**Comment by lvhan028** [comment]
Will it affect the vlm tokenizer?

**Comment by AllentDan** [comment]
> I did get the title "Set the add_special_token default to True" Is add_special_token==True indicated by *args and **kwargs?

Previously, we set `add_special_tokens` to False while huggingface AutoTokenizer set it default to True. So, I just deleted the `add_special_tokens=False` line in our codes.

**Review by lvhan028 — APPROVED** [review]

**Pull request merged**